src/rpc.cpp

```diff
@@ -624,6 +624,9 @@ RPCServer::~RPCServer() {
     m_objects.clear();
     m_callbacks.clear();
 
+    // Clear the Object Store App cache, to prevent instances from using the context which is going to be released.
+    app::App::clear_cached_apps();
```

Comment:
I am a bit unsure if this is the right place to call it.
I'm unaware of another place in our code that gets called when we're "unloading" our library, but we could clear the object store app cache when we initialize instead? https://github.com/realm/realm-js/blob/master/src/jsc/jsc_init.cpp#L42

Reply:
@blagoev @kneth ☝️ what do you think about that?

Reply:
I imagine the issue is when you reload the app, right? If so, I think it make sense to clear the cache as you shut down the RPC server.

Reply:
Correct. If it's not cleared before / after a reload object store might be reusing app instances that reference a context which has been garbage collected.

Reply:
:+1: